### PR TITLE
ps3joy_node root test and diagnostics 

### PR DESCRIPTION
**Pull request by alab288**
This pull request is based on the changes felix-kolbe's pull request contains after testing.

### CODE REVIEW

File: ps3joy/scripts/ps3joy.py [hunk 1]
```diff
@@ -408,9 +408,11 @@ def is_arg_with_param(arg, prefix):
...
             else:
                 print "Ignoring parameter: '%s'"%arg
-        if os.getuid() != 0:
-            print >> sys.stderr, "ps3joy.py must be run as root."
-            quit(1)
+        # TODO: Check for actual HW permissions, not root itself. Issue #53
+        # Error message hinting to root in case of failure is implemented already
+#        if os.getuid() != 0:
+#            print >> sys.stderr, "ps3joy.py must be run as root."
+#            quit(1)
```
Comment (jbohren-hbr) [inline ps3joy/scripts/ps3joy.py]: @alab288 You can remove these commented out lines.

File: ps3joy/README [hunk 2]
```diff
@@ -38,6 +38,15 @@ Starting the PS3 joystick
 
 rosrun ps3joy ps3joy.py
 
+Open a new terminal and reboot bluez and run joy with: 
+
+sudo systemctl restart bluetooth 
+rosrun joy joy_node
```
Comment (jbohren-hbr) [inline ps3joy/README]: Can you put these in code blocks?

See here for more info: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code

File: ps3joy/scripts/ps3joy_node.py [hunk 3]
```diff
@@ -181,7 +181,7 @@ def __init__(self, deamon, inactivity_timeout = float(1e3000)):
 
     def init_ros(self):
         try:
-            rospy.init_node('ps3joy',anonymous=True, disable_signals=True)        
+            rospy.init_node('ps3joy', anonymous=False, disable_signals=True)
```
Comment (jbohren-hbr) [inline ps3joy/scripts/ps3joy_node.py]: @alab288 You can change this back to `True` by default, but can you also check to see if this `anonymous` argument is overridden by roslaunch? i.e. create a simple roslaunch file where you specify the `name` of the node, and see if it overrides this `anonymous` flag.

File: ps3joy/scripts/ps3joy_node.py [hunk 4]
```diff
@@ -370,41 +372,53 @@ def __init__(self):
...
             return
         self.last_diagnostics_time = curr_time
+
+        ## compose diagnostics message
         diag = DiagnosticArray()
         diag.header.stamp = curr_time
-        #battery info
-        stat = DiagnosticStatus(name="Battery", level=DiagnosticStatus.OK, message="OK")
+        # battery info
+        stat = DiagnosticStatus(name='ps3joy'": Battery", level=DiagnosticStatus.OK, message="OK")
```
Comment (jbohren-hbr) [inline ps3joy/scripts/ps3joy_node.py]: @alab288 Can you look into the documentation for ros diagnostics and check to see if this is the correct pattern to use?

File: ps3joy/scripts/ps3joy_node.py [hunk 5]
```diff
@@ -563,6 +575,8 @@ def is_arg_with_param(arg, prefix):
 #                deamon = True
             else:
                 print "Ignoring parameter: '%s'"%arg
+        # TODO: Check for actual HW permissions, not root itself. Issue #53
+        # If the user does not have HW permissions indicate that ps3joy must be run as root
```
Comment (jbohren-hbr) [inline ps3joy/scripts/ps3joy_node.py]: You can remove this `#TODO` and associated comment.

File: ps3joy/scripts/ps3joy_node.py [hunk 6]
```diff
@@ -486,7 +499,6 @@ def listen(self, intr_sock, ctrl_sock):
                         if idev == cdev:
                             self.decoder.run(intr, ctrl)
                             print "Connection terminated."
-                            quit(0)
```
Comment (jbohren-hbr) [inline ps3joy/scripts/ps3joy_node.py]: @alab288 Are you sure this line should be removed?

**Comment by jbohren-hbr** [comment]
@alab288 Can you clean up the formatting of `README.md` so that it renders nicely?

For example, the formatting of the dependency list is shown as a section header here: https://github.com/alab288/joystick_drivers/blob/kinetic-devel/ps3joy/README.md

Take a look here for details on how to do this: https://help.github.com/articles/basic-writing-and-formatting-syntax/

**Comment by jbohren-hbr** [comment]
Also, have you tested this PR with hardware?

**Comment by alab288** [comment]
@jbohren-hbr yes I have tested this PR with the hardware.

**Pull request merged**